libs/blocks/locui/utils/miloc.js

```diff
@@ -86,24 +99,38 @@ export async function getProjectStatus() {
 export async function startSync() {
   setStatus('service', 'info', 'Syncing documents to Langstore.');
   const url = await getMilocUrl();
-  setExcelStatus('Sync to langstore/en.', '');
+  setExcelStatus('Sync to langstore/en', '');
   const opts = { method: 'POST' };
   const resp = await fetch(`${url}start-sync?project=${heading.value.projectId}`, opts);
   return resp.status;
 }
 
 export async function startProject({ skipSync }) {
   let url = await getMilocUrl();
-  setStatus('service', 'info', 'Starting project.');
+  setStatus('service', 'info', 'Starting project');
   const opts = { method: 'POST' };
   url = `${url}start-project?project=${heading.value.projectId}`;
   if (skipSync) url = `${url}&skipsync=true`;
   const resp = await fetch(url, opts);
-  if (resp.status === 201) setExcelStatus('Sent to localization service.', '');
+  if (resp.status === 201) setExcelStatus('Sent to localization service', '');
   setStatus('service');
   return resp.status;
 }
 
+export async function cancelProject() {
+  allowSyncToLangstore.value = false;
+  allowSendForLoc.value = false;
+  allowCancelProject.value = false;
+  let url = await getMilocUrl();
+  setStatus('service', 'info', 'Cancelling project');
+  const opts = { method: 'POST' };
+  url = `${url}cancel-project?project=${heading.value.projectId}`;
+  const resp = await fetch(url, opts);
+  if (resp.status === 200) setExcelStatus('Project cancelled', '');
```

Comment:
Hi @Sartxi - How are we handling the error scenario?

1. When the project Id is missing (which is difficult to replicate through the UI i think) - HTTP 400 - we might not need to address this
2. When the project is already in cancelled state (again, sort of difficult to happen through the UI as project status call will send the status and u will be hiding all the CTAs in that case) - HTTP 500 - we might not need to address this
3. Error in cancelling a GLaaS project and eventually cancelling the project - HTTP 500 - possible use case to handle

Reply:
@sabyamon I have added a handler for 500 errors that will display the error for the user in the status toast dialog